File: 1_prerequisite/python/hashMap.py

#----------------------------{ Synopsis of HashMaping in Csharp }----------------------------

# Create: d = {} or dict()
# Add/Update: d[key] = value
# Get: d[key] (KeyError) or d.get(key, default)
# Contains: key in d
# Remove: d.pop(key) or del d[key]
# Count: len(d)
# Keys/Values/Items: d.keys(), d.values(), d.items()
# Clear: d.clear()
# Update/merge: d.update(other_dict)
# Default-setting: d.setdefault(key, default)

#*/

#Defining maps...
d={}

#dding elements...
d["banana"]=5
d.update({'b':2})

#Printing Methods...
print(d.get("apple", 4)) 
print("banana" in d) #returns boolean
for k, v in d.items():
    print(k,"=>",v)


#Deletions Methods...
# d.pop("elppa") #returns error if key dosnt exist.
# del d["l"]  #returns error if key dosnt exist.
print(len(d))

File: 1_prerequisite/go/hashMap.go
package main

import "fmt"

/**----------------------------{ Synopsis of HashMaping in Csharp }----------------------------

Create: m := make(map[string]int) or var m map[string]int then m = make(...)
Add/Update: m[key] = value
Get + presence check: v, ok := m[key]
Delete: delete(m, key)
Length: len(m)
Iterate: for k, v := range m { ... }
Note: maps have no methods — use builtins len, delete, range.

*/


func main(){

	//Defining of Map
	m:=make(map[string]int) //defining: m := map[string]int{"apple": 0}
	
	//Insertion of in Map
	m["banana"]=2
	m["apple"]=5

	//Print a value of Map...
	if v, ok:= m["apple"];ok { fmt.Println(v) }
	//Printing Length...
	fmt.Println(len(m)) // 2
	//Printing all Values...
	for k, v := range m {
        fmt.Println(k, v)
    }

	//Removing of an ELement...
	delete(m, "apple")


}